Fix #5934: Display legend for external WMS layers

meyerlor opened the pull request:
I hope to help and fix the missing legends for external WMS services:

https://github.com/3liz/lizmap-web-client/issues/5934

External WMS layers were showing empty legend icons because QGIS Server
   returns empty icon fields in JSON GetLegendGraphic responses.

   - Added getLegendGraphicPNG() method to WMS class
   - Modified updateLayerTreeLayersSymbology() to detect external WMS layers
   - External WMS layers now fetch PNG legend and convert to base64
   - Normal layers continue using JSON format (no breaking changes)


I'm not sure if that is enough to get a proper legend from the WMS, i was not able to run proper tests, i just wanted to lay the groundwork to get WMS legends working (i hope it does not interfere with the recent changes which sped up the loading time so much! :smiley: )

rldhont commented:
Thanks @meyerlor 

Can you provide a printscreen with your fix ? I'm not sure that the GetLegendGraphic will be well displayed as an icon.

meyerlor commented:
I'm away from a computer the next three weeks, will get back at it as soon I'm back!

meyerlor commented:
I worked on it, now it's displaying for external wms the legend as PNG:
<img width="612" height="522" alt="grafik" src="https://github.com/user-attachments/assets/04554c94-c9bc-41fd-beec-081d76e4c2fc" />

The first layer is a rule based styled layer, here still the json is retrieved.

rldhont reviewed (changes requested):
Hi @meyerlor, 

First of all, thanks for your pull request. I suggest some improvements.

The main for me is to do not load the legend image and simulate a symbology but to enhance the Treeview component to display the legend as image when it is needed.

You will have to enhance the `this._layerTemplate` in `assets/src/components/Treeview.js`

meyerlor committed:
Fix #5934: Display legend for external WMS layers

External WMS layers were showing empty legend icons because QGIS Server
returns empty icon fields in JSON GetLegendGraphic responses.

- Add getLegendGraphicPNG() method to WMS class
- Detect external WMS layers via layerConfig.externalWmsToggle
- External WMS layers fetch PNG legend URL directly
- Update icon getter to handle URLs (http/https/relative paths)
- Set externalWmsToggle for WMS layers (not just EPSG:3857)
- Add LAYERTITLE=FALSE parameter to suppress redundant title

rldhont committed:
JS: Using image symbology for layer
